app/uk/gov/hmrc/fileupload/ApplicationLoader.scala

```diff
@@ -0,0 +1,301 @@
+    new CommandController(envelopeCommandHandler)
+  }
+
+  lazy val fileController = {
+    import play.api.libs.concurrent.Execution.Implicits._
+    val uploadBodyParser = UploadParser.parse(iterateeForUpload) _
+    new FileController(
+      withBasicAuth = withBasicAuth,
+      uploadBodyParser = uploadBodyParser,
+      retrieveFile = retrieveFile,
+      withValidEnvelope = withValidEnvelope,
+      handleCommand = envelopeCommandHandler)
+  }
+
+  lazy val adminController = {
+    new AdminController(getFileInfo = retrieveFileMetaData,
+      getChunks = fileChunksInfo)
+  }
+
+  lazy val transferController = {
+    val getEnvelopesByDestination = envelopeRepository.getByDestination _
+    val zipEnvelope = Zippy.zipEnvelope(find, retrieveFile) _
+    new TransferController(withBasicAuth, getEnvelopesByDestination, envelopeCommandHandler, zipEnvelope)
+  }
+
+  lazy val testOnlyController = {
+    new TestOnlyController(recreateCollections = List(eventStore.recreate, envelopeRepository.recreate, fileRepository.recreate, statsRepository.recreate))
+  }
+
+  lazy val routingController = {
+    new RoutingController(envelopeCommandHandler)
+  }
+
+  lazy val healthRoutes = new HealthRoutes(httpErrorHandler, new Provider[uk.gov.hmrc.play.health.AdminController] {
+    override def get() = new uk.gov.hmrc.play.health.AdminController(configuration)
+  })
+
+  lazy val appRoutes = new AppRoutes(httpErrorHandler, new Provider[EnvelopeController] {
+    override def get(): EnvelopeController = envelopeController
+  },
+    new Provider[FileController] {
+      override def get(): FileController = fileController
+    }, new Provider[EventController] {
+      override def get(): EventController = eventController
+    }, new Provider[CommandController] {
+      override def get(): CommandController = commandController
+    }, new Provider[AdminController] {
+      override def get(): AdminController = adminController
+    }
+  )
+
+  lazy val transferRoutes = new TransferRoutes(httpErrorHandler, new Provider[TransferController] {
+    override def get(): TransferController = transferController
+  })
+  lazy val routingRoutes = new RoutingRoutes(httpErrorHandler, new Provider[RoutingController] {
+    override def get(): RoutingController = routingController
+  })
+  lazy val metricsController = new MetricsController(new GraphiteMetricsImpl(applicationLifecycle, configuration))
+  lazy val adminRoutes = new AdminRoutes(httpErrorHandler, new Provider[MetricsController] {
+    override def get(): MetricsController = metricsController
+  })
+
+  lazy val prodRoutes = new Routes(httpErrorHandler, appRoutes, transferRoutes, routingRoutes,
+    healthRoutes, adminRoutes)
+
+  lazy val testRoutes = new testOnlyDoNotUseInAppConf.Routes(httpErrorHandler, testOnlyController, prodRoutes)
+
+  lazy val router: Router = if (configuration.getString("application.router").get == "testOnlyDoNotUseInAppConf.Routes") testRoutes else prodRoutes
```

Comment:
This will blow-up if application.router not set. We should avoid calls to get on an Option.